services/analytics/src/main/java/com/wedriveu/services/analytics/entity/AnalyticsStoreImpl.java

```diff
@@ -86,4 +89,15 @@ public void clear() {
             Log.error(TAG, CLEAR_ERROR, e);
         }
     }
+
+    @Override
+    public VehicleCounter getVehicleCounter() {
+        try {
+            VehiclesCounterAlgorithm vehiclesCounterAlgorithm = new VehiclesCounterAlgorithm();
```

Comment:
So, if i want to change the `VehiclesCounterAlgorithm` i have to change the `AnalyticsStore` implementation? Wouldn't it be simpler to treat that algorithm as a strategy and pass its interface (or trait) through the `AnalyticsStoreImpl` constructor from outside? Then you can use it here.